nucypher/blockchain/eth/interfaces.py

```diff
@@ -562,9 +562,14 @@ def sign_and_broadcast_transaction(self,
         #
 
         # TODO: Show the USD Price:  https://api.coinmarketcap.com/v1/ticker/ethereum/
-        max_tx_price = transaction_dict['maxFeePerGas']
-        max_priority_price = transaction_dict['maxPriorityFeePerGas']
-        max_price = max_tx_price + max_priority_price
+        
+        if transaction_dict.get('maxFeePerGas') and transaction_dict.get('maxPriorityFeePerGas'):
+            # TODO:  this probably only needs to be here to support Ropsten?
+            max_tx_price = transaction_dict['maxFeePerGas']
+            max_priority_price = transaction_dict['maxPriorityFeePerGas']
+            max_price = max_tx_price + max_priority_price
```

Comment:
I don't think this is right, the TX price will never be higher than the `maxFeePerGas` (see https://web3py.readthedocs.io/en/stable/web3.eth.html#web3.eth.Eth.send_transaction).

The thing is that I don't see how web3py populates `maxFeePerGas` or `maxPriorityFeePerGas` to give you a complete opinion here, since both fields are optional, and to make it more complex, depend on the ETH provider. In principle, the simplest approach to make sure your TX will be mined is to just hard-code a max priority fee per gas (e.g., 1 or 1.5 gwei), and call it a day, but I haven't tried this with web3py (since they use to do a lot of weird processing of TX data on their side when building and sending).

Now, this code tries to estimate the actual cost of the TX. In that case, instead of querying `maxFeePerGas`, we should get the current `baseFee` and the estimated cost will be `baseFee + maxPriorityFeePerGas`

Reply:
> The thing is that I don't see how web3py populates maxFeePerGas or maxPriorityFeePerGas 

Reading through the source code, I don't see an RPC call (unless I'm missing it?)  IT appears to just be a local gas estimation.  The lowest level call from the nucypher codebase is to `buildTransaction` which includes gas estimation.  Perhaps our usage is somehow signals a legacy transaction.

> In principle, the simplest approach to make sure your TX will be mined is to just hard-code a max priority fee per gas (e.g., 1 or 1.5 gwei), and call it a day, but I haven't tried this with web3py (since they use to do a lot of weird processing of TX data on their side when building and sending).

This particular block of code isn't used to populate a transaction to be broadcasted, it's just a console message for the user's convenience.  If it's too much trouble to debug this block, we can simply remove it with no consequences.

> Now, this code tries to estimate the actual cost of the TX. In that case, instead of querying maxFeePerGas, we should get the current baseFee and the estimated cost will be baseFee + maxPriorityFeePerGas

Since `maxFeePerGas = baseFee + maxPriorityFeePerGas` and I expect the ransaction fields to be filled by web3 for this type of transaction, I think we can juts set `max_price` here to `transaction_dict['maxFeePerGas']` and call it a day.